tests/integration/thrift_tests.py

```diff
@@ -406,6 +434,70 @@ def example(self, context):
 
         assert handler.edge_context == FakeEdgeContextFactory.DECODED_CONTEXT
 
+    def test_budget_header_pool_timeout(self):
+        """Test that the budget header is set in the headers with the pool timeout."""
+        budget = 100.0
+
+        class Handler(TestService.Iface):
+            def example(self, context):
+                self.context = context
+                return True
+
+        handler = Handler()
+
+        span_observer = mock.Mock(spec=SpanObserver)
+        with serve_thrift(handler, TestService) as server:
+            with baseplate_thrift_client(server.endpoint, TestService, span_observer) as context:
```

Comment:
this test always takes me a few looks to figure out why do we define the budget as `100.0` above and assert that the budget equals to `str(1000)`. my suggestions are:

1. Here explicitly add `timeout = "1 second"` to `baseplate_thrift_client` so we are not relying on the default value
2. rename the `budget` variable on line 439 to `pool_timeout_seconds`.